### PR TITLE
Promise refactor

### PR DESCRIPTION
**Pull request by cbandrow-zz**
## What's this PR do?

Took a long look and study at Promises tonight. Originally, the ImageImport component was using a promise to ensure that the state was set after the resolve of the filer-reader. Upon investigation, this was not needed. 

However, upon further research, the promise within the App retrieving that data value from ImageImport is very well needed to ensure all data is stored before making an API call. 

This PR removes the promise in ImageImport and refactors the promise in App to make use of the Resolve/Reject structure. 

## Learning

Quality info about the Promises and the FileReader can be found here. 

[Google JS Promises](https://developers.google.com/web/fundamentals/getting-started/primers/promises)
[MDN Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
[File Reader Info](https://www.html5rocks.com/en/tutorials/file/dndfiles/)
[Promise.js (side note, best promise resource I've found)](https://www.promisejs.org/)

## Where should the reviewer start?

Feel free to take a look at the function starting on line 23 of ImageImport component, and the function starting at line 34 of App. 

## Context

The ImageImport function of handleImageUpload is used to convert an image into a base64 encoded url, which is really just a big friggin string. It passes that value to state. Upon submit, it passes that big friggin string to App, where it sets state and prepares to send the value off to the API.

When it get's to App, I was noticing that if immediately called, setting the state of App with the big friggin string wouldn't be set immediately, and would cause problems. By wrapping it in a custom promise, the promise allows itself to be evaluated and resolved or rejected as soon as state is actually set. 

## What gif best describes this PR or how it makes you feel?

promises

### CODE REVIEW

File: app/bundle.js [hunk 1]
```diff
@@ -65,7 +65,7 @@
 /******/ 	}
 /******/ 	
 /******/ 	var hotApplyOnUpdate = true;
```
Comment (letakeane) [inline app/bundle.js:67]: Can these "var"s be refactored into "const"s or "let"s?

File: app/components/ImageImport/ImageImport.js [hunk 2]
```diff
@@ -26,22 +26,13 @@ export default class ImportImage extends Component {
...
-       resolve("");
-     }, 200);
-   });
-   imagePromise.then(()=>{
-     console.log("Image conversion promise resolved.")
-     console.log(this.state)
-   })
+   reader.readAsDataURL(file)
+   reader.onloadend = () => {
+      this.setState({
```
Comment (letakeane) [inline app/components/ImageImport/ImageImport.js:31]: This looks a lot cleaner than the setTimeout!

File: app/bundle.js (hunk 1 above)
Comment (cbandrow-zz) [reply app/bundle.js:67]: They're bundle stuffs, so i'm not sure.

File: app/components/App.js [hunk 3]
```diff
@@ -34,22 +34,28 @@ export default class App extends Component {
   handleImageData(inputState){
     let content
     let statePromise = new Promise((resolve, reject)=>{
-      console.log('loading...')
```
Comment (lauraturk) [inline app/components/App.js]: do you need this console.log still?

File: app/components/App.js [hunk 4]
```diff
@@ -34,22 +34,28 @@ export default class App extends Component {
...
         loadingStatus: true,
         imagePreviewUrl: inputState.imagePreviewUrl,
       })
-      setTimeout(function(){
-        resolve('promise resolved');
-      }, 1100);
+
+      setTimeout(()=>{
+        if(this.state.imagePreviewUrl && this.state.loadingStatus){
+          console.log('loading...')
```
Comment (lauraturk) [inline app/components/App.js:47]: Do you want to keep this console.log?

File: app/components/ImageImport/ImageImport.js (hunk 2 above)
Comment (lauraturk) [reply app/components/ImageImport/ImageImport.js:31]: I like breaking this out of the image component, it looks a lot better and the onloadend was essentially doing the same as the timeout, right? Looking good! 

You might consider changing your console log errors on line 58 of App into something the user will see?

**Comment by lauraturk** [comment]
Left a few comments around UI for errors and console.logs.  Looks like a great bit of refactoring!

File: app/components/App.js (hunk 4 above)
Comment (cbandrow-zz) [reply app/components/App.js:47]: I don't think I need to keep them in there, I'll remove them upon the next refactor.

**Pull request merged**